src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -21,39 +24,114 @@ object UserService {
     Props(userServiceConstructor(userInfo))
   }
 
-  def constructor(dataSource: DataSource, googleServicesDAO: GoogleServicesDAO, containerDAO: GraphContainerDAO)(userInfo: UserInfo)(implicit executionContext: ExecutionContext) =
-    new UserService(userInfo, dataSource, googleServicesDAO, containerDAO)
+  def constructor(dataSource: DataSource, googleServicesDAO: GoogleServicesDAO, containerDAO: GraphContainerDAO, userDirectoryDAO: UserDirectoryDAO)(userInfo: UserInfo)(implicit executionContext: ExecutionContext) =
+    new UserService(userInfo, dataSource, googleServicesDAO, containerDAO, userDirectoryDAO)
 
   sealed trait UserServiceMessage
   case class SetRefreshToken(token: UserRefreshToken) extends UserServiceMessage
   case object GetRefreshTokenDate extends UserServiceMessage
   case object CreateUser extends UserServiceMessage
+  case class GetUserStatus(userRef: RawlsUserRef) extends UserServiceMessage
+  case class EnableUser(userRef: RawlsUserRef) extends UserServiceMessage
+  case class DisableUser(userRef: RawlsUserRef) extends UserServiceMessage
 }
 
-class UserService(userInfo: UserInfo, dataSource: DataSource, googleServicesDAO: GoogleServicesDAO, containerDAO: GraphContainerDAO)(implicit executionContext: ExecutionContext) extends Actor {
+class UserService(protected val userInfo: UserInfo, dataSource: DataSource, protected val gcsDAO: GoogleServicesDAO, containerDAO: GraphContainerDAO, userDirectoryDAO: UserDirectoryDAO)(implicit protected val executionContext: ExecutionContext) extends Actor with AdminSupport with FutureSupport {
```

Comment:
no particular reason, can't be private because it is required by AdminSupport and public did not seem right